src/TwigExtension.php

```diff
@@ -36,80 +45,56 @@ public function is_instanceof($value, string $type): bool
-   *   Whether to check for entity access permission (default TRUE).
    * @param  int  $limit
    *   Restrict to number of results. Capped at no more than 100.
-   * @param  null  $langcode
-   *   (optional) For which language the entity should be rendered, defaults to
-   *   the current content language.
    *
    * @return null|array
    *   An array of render arrays for the entities found, or NULL if the entity does not exist.
    */
-  public function sbf_entity_ids_by_label(string $label, string $entity_type, string $bundle_identifier = '', $check_access = TRUE, $limit = 1, $langcode = NULL): ?array {
-    $label = \Drupal::database()->escapeLike($label);
-    $limit = min($limit, 100);
-    /** @var \Drupal\Core\Entity\Query\QueryInterface $query */
-    try {
-      switch($entity_type) {
-        case 'node':
-          $query = \Drupal::entityTypeManager()->getStorage('node')->getQuery();
-          $query->condition('title', $label, 'LIKE')
-            ->accessCheck($check_access)
-            ->range(0,$limit);
-          if($bundle_identifier) {
-            $query->condition('type', $bundle_identifier);
-          }
-          $ids = $query->execute();
-          break;
-        case 'taxonomy_term':
-          $query = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->getQuery();
-          $query->condition('name', $label, 'LIKE')
-            ->accessCheck($check_access)
-            ->range(0,$limit);
-          if($bundle_identifier) {
-            $query->condition('vid', $bundle_identifier);
-          }
-          $ids = $query->execute();
-          break;
-        case 'group':
-          $query = \Drupal::entityTypeManager()->getStorage('group')->getQuery();
-          $query->condition('label', $label, 'LIKE')
-            ->accessCheck($check_access)
-            ->range(0,$limit);
-          if($bundle_identifier) {
-            $query->condition('type', $bundle_identifier);
-          }
-          $ids = $query->execute();
-          break;
-        case 'user':
-          $query = \Drupal::entityTypeManager()->getStorage('user')->getQuery();
-          $query->condition('name', $label, 'LIKE')
-            ->accessCheck($check_access)
-            ->range(0,$limit);
-          $ids = $query->execute();
-          break;
-      }
-    }
+  public function entityIdsByLabel(
+    string $label,
+    string $entity_type,
+    string $bundle_identifier = '',
+    $limit = 1
+  ): ?array {
+    $fields = [
+      'node' => ['title', 'type'],
+      'taxonomy_term' => ['name', 'vid'],
+      'group' => ['label', 'type'],
+      'user' => ['name', NULL],
+    ];
+    $label_field = $fields[$entity_type][0] ?? NULL;
+    if ($label_field) {
+      $bundle_field = $fields[$entity_type][1] ?? NULL;
+      $limit = min($limit, 100);
+      /** @var \Drupal\Core\Entity\Query\QueryInterface $query */
+      try {
+        $query = \Drupal::entityTypeManager()->getStorage('node')->getQuery();
```

Comment:
@patdunlavey this is looking so good. Thanks for addressing my code review, I really appreciate it. You may be working on this already,  but just in case because I will forget it mysql later:

`getStorage('node')` should be after your refactor `getStorage($entity_type')`

Let me know when you want me to do some testing. Again, thanks and good work :Clarkson

Reply:
Oops! How'd that sneak back in there!

Reply:
If you're inclined to try testing, I think it's ready for that @DiegoPino .